lib/govuk_publishing_components/presenters/step_by_step_nav_helper.rb

```diff
@@ -92,10 +92,24 @@ def get_list_element(style)
         style == "choice" ? "ul" : "ol"
       end
 
-      def link_href(active, href)
+      def link_href(active, href, parameters = {})
+        external_url = href.start_with?('http')
+        href = external_url ? href : update_href_with_parameters(href, parameters)
         active ? "#content" : href
       end
 
+      def update_href_with_parameters(href, parameters)
+        if parameters.any?
+          uri = URI.parse(href)
+          query_params = uri.query.present? ? CGI.parse(uri.query) : {}
+          query_params = query_params.merge(parameters)
```

Comment:
Following on from some of the comments above, you could just merge the step_nav id here.
e.g. `query_params.merge("step-by-step-nav" => @options[:step_nav_content_id])`